oxo.py

```diff
@@ -1,29 +1,159 @@
 class OxoBoard:
+
     def __init__(self):
         """ The initialiser. Initialise any fields you need here. """
-        raise NotImplementedError("TODO: implement __init__")
+        self.board_rows = 3
+        self.board_columns = 3
+        self.board_array = []
+        # create the board array
+        for x in range(self.board_rows):
+            self.board_array.append([])
+
+            for y in range(self.board_columns):
+                self.board_array[x].append(0)
+
+        self.show()
 
     def get_square(self, x, y):
         """ Return 0, 1 or 2 depending on the contents of the specified square. """
-        raise NotImplementedError("TODO: implement get_square")
+
+        square_content = self.board_array[x][y]
+        return square_content
 
     def set_square(self, x, y, mark):
         """ If the specified square is currently empty (0), fill it with mark and return True.
             If the square is not empty, leave it as-is and return False. """
-        raise NotImplementedError("TODO: implement set_square")
+
+        if self.board_array[x][y] == 0:
+            self.board_array[x][y] = mark
+            return True
+        else:
+            return False
 
     def is_board_full(self):
         """ If there are still empty squares on the board, return False.
             If there are no empty squares, return True. """
-        raise NotImplementedError("TODO: implement is_board_full")
+
+        full_columns = 0
+
+        # check if all of the columns are full
+        for x in range(self.board_rows):
+            if 0 in self.board_array[x]:
+                return False
+            else:
+                full_columns += 1
+        if full_columns == self.board_rows:
+            return True
+
+    def check_horizontal(self):
+        """Check if a player has connected a row horizontally."""
+
+        for x in range(self.board_rows):
+            player1_score = 0
+            player2_score = 0
+            for y in range(self.board_columns):
+                if 1 == self.board_array[y][x]:
+                    player1_score += 1
+                else:
+                    player1_score = 0
+                if player1_score == self.board_columns:
+                    return 1
+
+                if 2 == self.board_array[y][x]:
+                    player2_score += 1
+                else:
+                    player2_score = 0
+                if player2_score == self.board_columns:
+                    return 2
+
+        return 0
+
+    def check_diagonal(self):
+        """Check if a player has connected a row diagonally."""
+
+        connected_o_1 = 0
+        connected_x_1 = 0
```

Comment:
You have used `player1` and `player2` in `check_horizontal`, but `o` and `x` here. Better to be consistent